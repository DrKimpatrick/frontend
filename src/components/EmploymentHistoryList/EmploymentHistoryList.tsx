import React, { FC, Fragment, useEffect } from 'react';
import { withRouter } from 'react-router-dom';
import ArrowBackTwoToneIcon from '@material-ui/icons/ArrowBackTwoTone';
import ArrowRightAltTwoToneIcon from '@material-ui/icons/ArrowRightAltTwoTone';
import AddCircleOutlineOutlinedIcon from '@material-ui/icons/AddCircleOutlineOutlined';
import NavBar from 'components/Layout/NavBar/NavBar';

type props = {};
const EmploymentHistoryList: FC<props> = (props: any) => {

  useEffect(() => {
    const token = localStorage.getItem('token');
    if (!token) {
      props.history.push('/login');
    }
  });
  return (
    <Fragment>
      <NavBar />
      <section className="employment-history-list-section w-1/3 m-auto text-textGray">
        <div className="flex relative h-auto my-8">
          <div
            className="back-arrow cursor-pointer"
            onClick={() => props.history.push('/employment-history')}
          >
            <ArrowBackTwoToneIcon />
          </div>
          <h1 className="font-bold text-xl title">Employment History</h1>
        </div>

        <div className="text-textGray mt-8 border border-borderGray card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Google, VP of Engineering
          </div>
          <div className="py-3 px-4 card-content">March, 2020 - Current</div>
        </div>

        <div className="text-textGray mt-4 border border-borderGray card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Facebook, Director of Engineering
          </div>
          <div className="py-3 px-4 card-content">March, 2018 - March 2019</div>
        </div>

        <div className="text-textGray mt-4 border border-borderGray card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Twitter, Technical Lead
          </div>
          <div className="py-3 px-4 card-content">March, 2017 - March 2018</div>
        </div>

        <div className="text-textGray mt-4 border border-borderGray card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Snapchat, Senior Software Engineer
          </div>
          <div className="py-3 px-4 card-content">
            March, 2016 - March, 2017
          </div>
        </div>

        <div className="text-textGray mt-4 border border-borderGray card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            AR Devs, Mid Level Software Engineer
          </div>
          <div className="py-3 px-4 card-content">March, 2015 - March 2016</div>
        </div>

        <div className="text-textGray mt-4 border border-borderGray card">
          <div className=" text-white font-bold py-3 px-4 card-title">
            Bank of America, Junior Software Engineer
          </div>
          <div className="py-3 px-4 card-content">
            March, 2014 - March, 2015
          </div>
        </div>

        <div className="flex justify-center mt-12">
          <button
            data-testid="next-button"
            className="add-employer-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-64 rounded-sm shadow flex justify-around"
          >
            <span className="">Add another employer</span>
            <AddCircleOutlineOutlinedIcon />
          </button>
        </div>
        <div className="flex justify-center mt-12">
          <button
            data-testid="next-button"
            className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
            onClick={() => props.history.push('/user/dashboard')}
          >
            <span className="">Next</span> <ArrowRightAltTwoToneIcon />
          </button>
        </div>
      </section>
    </Fragment>
  );
};

export default withRouter(EmploymentHistoryList);
