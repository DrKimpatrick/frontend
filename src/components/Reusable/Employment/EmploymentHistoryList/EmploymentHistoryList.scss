.employment-history-list-section {
  width: 33%;
  .title {
    font-size: 22px;
    color: #747474;
    text-decoration: none solid rgb(116, 116, 116);
  }

  .back-arrow {
    position: absolute;
    color: #747474;
    left: -43px;
    top: 0px;
    font-size: 22px;
  }

  .input-height {
    height: 3rem;
    border-radius: 4px;
    color: #747474;
    font-size: 14px;
  }

  .card {
    border-radius: 2px;

    .card-title {
      font-size: 15px;
      background-color: #38727e;
    }

    .card-content {
      color: #303030;
      font-size: 14px;
    }
  }

  .add-employer-btn {
    height: 41px;
    align-items: center;
    background-color: #3e8074;
  }
}

@media screen and (max-width: 786px) {
  .employment-history-list-section {
    width: 50%;
    margin-bottom: 100px !important;

    .back-arrow {
      display: none;
    }

    .title {
      width: 100%;
      text-align: center;
    }
  }
}

@media (max-width: 500px) {
  .employment-history-list-section {
    width: 90%;
  }
}
