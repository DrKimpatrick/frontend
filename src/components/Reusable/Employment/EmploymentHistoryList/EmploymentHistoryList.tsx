/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React, { FC, useEffect } from 'react';
import { withRouter, useHistory } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import ArrowBackTwoToneIcon from '@material-ui/icons/ArrowBackTwoTone';
import ArrowRightAltTwoToneIcon from '@material-ui/icons/ArrowRightAltTwoTone';
import AddCircleOutlineOutlinedIcon from '@material-ui/icons/AddCircleOutlineOutlined';
import NavBar from 'components/Reusable/Layout/NavBar/NavBar';
import { RootState } from 'redux/store';
import { listEmployments } from 'redux/actions/employment';
import { MainBackground } from 'components/Reusable/Layout/MainBackground';
import './EmploymentHistoryList.scss';

const EmploymentHistoryList: FC = () => {
  const dispatch = useDispatch();

  const history = useHistory();

  useEffect(() => {
    listEmployments()(dispatch);
  }, [dispatch]);

  const reducer = useSelector((state: RootState) => {
    const { loading, errors, employments } = state.employments;
    const { message } = state.messages;
    return { message, loading, errors, employments };
  });

  const { employments } = reducer;

  if (!employments) {
    return null;
  }
  return (
    <>
      <NavBar />
      <section className="employment-history-list-section w-1/3 m-auto text-textGray">
        <div className="flex relative h-auto my-8">
          <div
            className="back-arrow cursor-pointer"
            onClick={() => history.push('/employment-history')}
          >
            <ArrowBackTwoToneIcon />
          </div>
          <h1 className="font-bold text-xl title">Employment History List</h1>
        </div>

        {employments &&
          employments.map((item, i) => (
            <div
              className="text-textGray mt-4 border border-borderGray card"
              key={i}
            >
              <div className=" text-white font-bold py-3 px-4 card-title">
                {item.companyName} , {item.title}
              </div>
              <div className="py-3 px-4 card-content">
                {new Date(item.startDate).toDateString()}{' '}
                {item.endDate && ` to ${new Date(item.endDate).toDateString()}`}
              </div>
            </div>
          ))}

        <div className="flex justify-center mt-12">
          <button
            data-testid="next-button"
            className="add-employer-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-64 rounded-sm shadow flex justify-around"
            onClick={() => history.push('/recent-employer')}
            type="button"
          >
            <span className="">Add another employer</span>
            <AddCircleOutlineOutlinedIcon />
          </button>
        </div>
        <div className="flex justify-center mt-12">
          <button
            data-testid="next-button"
            className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
            onClick={() => history.push('/add-education')}
            type="button"
          >
            <span className="">Next</span> <ArrowRightAltTwoToneIcon />
          </button>
        </div>
      </section>
      <MainBackground />
    </>
  );
};

export default withRouter(EmploymentHistoryList);
