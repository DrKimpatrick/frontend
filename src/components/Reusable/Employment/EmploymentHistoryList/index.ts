import EmploymentHistoryList from './EmploymentHistoryList';

export { EmploymentHistoryList };
