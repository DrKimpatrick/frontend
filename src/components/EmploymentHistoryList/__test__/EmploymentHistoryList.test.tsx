import { cleanup } from '@testing-library/react';
import React from 'react';
import ReactDom from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import EmploymentHistoryList from '../EmploymentHistoryList';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);
const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush
  })
}));

describe('`EmploymentHistoryList` component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(
      <Router>
        <EmploymentHistoryList />
      </Router>,
      div
    );
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <EmploymentHistoryList />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
