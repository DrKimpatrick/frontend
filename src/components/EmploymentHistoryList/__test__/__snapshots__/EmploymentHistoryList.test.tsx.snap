// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`EmploymentHistoryList\` component should create snapshot 1`] = `null`;
